dough-data/src/test/java/io/github/bakedlibs/dough/collections/TestCopyUtils.java

```diff
@@ -0,0 +1,109 @@
+package io.github.bakedlibs.dough.collections;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+class TestCopyUtils {
+
+    private static <T> void assertHaveEqualElements(@Nonnull Collection<T> c1, @Nonnull Collection<T> c2) {
+        Assertions.assertEquals(c1.size(), c2.size());
+        Assertions.assertTrue(c1.containsAll(c2));
+    }
+
+    private static <K, V> void assertHaveEqualElements(@Nonnull Map<K, V> m1, @Nonnull Map<K, V> m2) {
+        Assertions.assertEquals(m1.size(), m2.size());
+        for (Map.Entry<K, V> entry : m1.entrySet()) {
+            Assertions.assertEquals(m2.get(entry.getKey()), entry.getValue());
+        }
+    }
+
+    private static <T> void assertHaveClonedElements(@Nonnull Collection<T> c1, @Nonnull Collection<T> c2) {
+        Assertions.assertEquals(c1.size(), c2.size());
+        for (T t1 : c1) {
+            for (T t2 : c2) {
+                // If they are the same, it did not clone and thus we fail.
+                Assertions.assertNotSame(t1, t2);
+                if (t1.equals(t2)) {
+                    // If they are equal, the element has been cloned, thus, we break and check the next element.
+                    break;
+                }
+            }
+        }
+    }
```

Comment:
Collections can be unordered, making this fail. This implementations only works for ordered/sorted collections, aka Lists.

Reply:
Why would this fail if it was unordered? Maybe I'm just very sleepy but I can't seem to think of why this would fail if it wasn't ordered